src/orbitalMechanics.cpp

```diff
@@ -0,0 +1,405 @@
+      //v1Vec.resize(nMax * 2 + 1);
+      //std::vector<arma::Col<double>::fixed<3>> v2Vec;
+      //v2Vec.resize(nMax * 2 + 1);
+      std::vector<std::pair<arma::Col<double>::fixed<3>, arma::Col<double>::fixed<3>>> velocitiesVector;
+      velocitiesVector.reserve(nMax * 2 + 1);
+      //std::vector<int> itersVec;
+      //itersVec.resize(nMax * 2 + 1);
+      std::vector<double> xVector;
+      xVector.reserve(nMax * 2 + 1);
+
+      // 3 - We may now find all solutions in x,y
+      // 3.1 0 rev solution
+      // 3.1.1 initial guess
+      if (T >= T00) {
+        xVector.at(0) = -(T - T00) / (T - T00 + 4.0);
+      } else if (T <= T1) {
+        xVector.at(0) = T1 * (T1 - T) / (2.0 / 5.0 * (1.0 - lambdaSquare * lambdaCubic) * T) + 1.0;
+      } else {
+        xVector.at(0) = std::pow((T / T00), 0.69314718055994529 / std::log(T1 / T00)) - 1.0;
+      }
+
+      // 3.1.2 Householder iterations
+      //itersVec[0] = brent(...); //need??? or TODO -> householder(T, xVector[0], 0.0, 1e-5, 15);
+
+      // 3.2 multi rev solutions
+      double tmp;
+      for (int i = 1; i < nMax + 1; ++i) {
+        //3.2.1 left Householder iterations
+        tmp = std::pow((i * arma::datum::pi + arma::datum::pi) / (8.0 * T), 2.0 / 3.0);
+        xVector.at(2 * i - 1) = (tmp - 1.0) / (tmp + 1.0);
+        //itersVec[2 * i - 1] = brent(...); //need??? or TODO -> householder(T, xVector[2*i-1], i, 1e-8, 15);
+
+        //3.2.2 right Householder iterations
+        tmp = std::pow((8.0 * T) / (i * arma::datum::pi), 2.0 / 3.0);
+        xVector.at(2 * i) = (tmp - 1.0) / (tmp + 1.0);
+        //itersVec[2 * i] = brent(...); //need??? or TODO -> householder(T, xVector[2 * i], i, 1e-8, 15);
+      }
+
+      /// 4 - For each found x value we reconstruct the terminal velocities
+      double gamma = std::sqrt(muSun * lengthSumWeight / 2.0);
+      double rho = (departurePositionVectorLength - arrivalPositionVectorLength) / flightVectorLength;
+      double sigma = std::sqrt(1 - std::pow(rho, 2.0));
+
+      double vt, vr1, vt1, vr2, vt2, y;
+
+      for (size_t i = 0; i < xVector.size(); ++i) {
+        y = std::sqrt(1.0 - lambdaSquare + lambdaSquare * std::pow(xVector.at(i), 2.0));
+        vr1 = gamma * ((lambda * y - xVector.at(i)) - rho * (lambda * y + xVector.at(i))) / departurePositionVectorLength;
+        vr2 = -gamma * ((lambda * y - xVector.at(i)) + rho * (lambda * y + xVector.at(i))) / arrivalPositionVectorLength;
+
+        vt = gamma * sigma * (y + lambda * xVector.at(i));
+        vt1 = vt / departurePositionVectorLength;
+        vt2 = vt / arrivalPositionVectorLength;
+
+        //v1Vec.at(i) = vr1 * departurePositionNormalised + vt1 * it1;
+        //v2Vec.at(i) = vr2 * arrivalPositionNormalised + vt2 * it2;
+
+        velocitiesVector.push_back(std::make_pair(vr1 * departurePositionNormalised + vt1 * it1, vr2 * arrivalPositionNormalised + vt2 * it2));
+      }
+
+      return velocitiesVector;
+    }
+
+    arma::Col<double>::fixed<3> dTdx(
```

Comment:
I would note a lot about this function, but as it should be unused/removed, I ignore this for now.